nov.el

```diff
@@ -797,6 +799,38 @@ Saving is only done if `nov-save-place-file' is set."
 (add-hook 'nov-mode-hook 'nov-add-to-recentf)
 (add-hook 'nov-mode-hook 'hack-dir-local-variables-non-file-buffer)
 
+
+(defun nov--find-file (file index position)
+  "Open FILE(nil means current buffer) in nov-mode and go to the specified INDEX and POSITION."
+  (when file
+    (find-file file))
+  (unless (eq major-mode 'nov-mode)
+    (nov-mode))
+  (when (not (nov--index-valid-p nov-documents index))
+    (error "Invalid documents index"))
+  (setq nov-documents-index index)
+  (nov-render-document)
+  (goto-char point))
+
+;; Bookmark Integration
```

Comment:
You must `(require 'bookmark)` here. Maybe move all the requires to the beginning since the first bookmark variable is used before that.

Reply:
Yes, I will do that